packages/thrift-client/src/main/connections/HttpConnection.ts

```diff
@@ -217,56 +214,47 @@ export class HttpConnection extends Core.ThriftConnection<RequestOptions> {
                 : this.url
 
         // Merge user options with required options
-        const requestOptions: RequestOptions & UrlOptions = Core.overlayObjects(
-            this.requestOptions,
-            options,
+        const requestOptions: RequestOptions = mergeOptions(
+            mergeOptions(this.requestOptions, options),
```

Comment:
doublechecking `encoding: null, // Needs to be explicitly set to null to get Buffer in response body` is not needed with new cross-fetch

Reply:
Right, this seems to be a `request`-specific problem, it works fine in both the unit tests and in my testing with Athena UI. There's no option in `fetch` or `node-fetch` that would provide the same functionality. `request` had a ton of options (see https://github.com/request/request#requestoptions-callback) and `fetch` is much simpler, the `Response` can be read in a few different ways and we always read it as binary data using `arrayBuffer`.